requests_ntlm/requests_ntlm.py

```diff
@@ -86,8 +87,8 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
         # get the challenge
         auth_header_value = response2.headers[auth_header_field]
 
-        ntlm_header_value = list(filter(
-            lambda s: s.startswith('NTLM '), auth_header_value.split(',')
+        ntlm_header_value = list(filter(lambda s: s.startswith('NTLM '),
+                                        map(string.lstrip, auth_header_value.split(','))
```

Comment:
With some correction (missing  `for val` and the final strip as earlier):

```
ntlm_header_value = [
    s for s in (val.lstrip() for val in auth_header_value.split(','))
    if s.startswith('NTLM ')
][0].strip()
```

(And I did verify that it's faster on simple data)

But note that I am not currently using requests-ntlm and does not have the setup for testing this properly myself anymore.